proxy/http2/Http2ConnectionState.h

```diff
@@ -146,8 +148,11 @@ class Http2ConnectionState : public Continuation
     local_hpack_handle = nullptr;
     delete remote_hpack_handle;
     remote_hpack_handle = nullptr;
-    delete dependency_tree;
-    dependency_tree  = nullptr;
+
+    if (this->dependency_tree) {
+      delete dependency_tree;
+      dependency_tree = nullptr;
```

Comment:
The only reason for this check, I assume, is to only set dependency_tree=nullptr (it's always safe / no-op to delete a nullptr). If we don't need to set dependency_tree to nullptr, I'd just get rid of the conditional.

Reply:
+1 on removing the conditional

Reply:
Sounds reasonable. Will do.